sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -345,11 +347,18 @@ def from_estimator(
             np.linspace(x0_min, x0_max, grid_resolution),
             np.linspace(x1_min, x1_max, grid_resolution),
         )
-        if hasattr(X, "iloc"):
+        if _is_pandas_df(X):
             # we need to preserve the feature names and therefore get an empty dataframe
             X_grid = X.iloc[[], :].copy()
             X_grid.iloc[:, 0] = xx0.ravel()
             X_grid.iloc[:, 1] = xx1.ravel()
+        elif _is_polars_df(X):
+            X_grid = X.drop(X.columns).with_columns(
+                **{
+                    column: series
+                    for column, series in zip(X.columns, (xx0.ravel(), xx1.ravel()))
+                }
+            )
```

Comment:
I noticed adapter was introduced in PR #27315. Can we use it here to create a uniform solution for Pandas and Polars, for example `adapter.create_container`?

Reply:
Yes indeed! This is a bit nicer.